### ISSUE TITLE
BSgenome

### ISSUE THREAD

**Issue by 1kabutar** [Aufiero/circRNAprofiler#15]
Hi Simona,

Thanks for actively improving the package as it has been a breeze so far to use even for the large non-model genome (>10gb) that I am working with.
I am stuck with the BSgenome as it has no support for 64bit large genome yet, I can't really do things such as retrieve sequence or anything that requires for e.g. targetsFTS_gr, snp, mirna, repeat elements, etc. 

Is there a workaround or alternative to BSgenome that I can use?

Best,
Ketan

**Comment by simoauf** [comment]
Hi 1kabutar, I am not sure that I understand your problem with BSgenome. Can you explain me better?

**Comment by Aufiero** [comment]
Hi 1kabutar, I am not sure that I understand your problem with BSgenome. Can you explain me better?

**Comment by 1kabutar** [comment]
Understood, so let me try again, all aspects of the package which rely on sequence extraction, I cannot perform as the sequence extraction relies on BSgenome.
I have tried to create a BSgenome package using 'forgeBSgenomeDataPkg()' to substitute this for "BSgenome.Hsapiens.UCSC.hg19" that is used in the doc/tutorial - but this has been unsuccessful due to the large size of the genome I'm working with.
I'm wondering then if there is an alternative to BSgenome? Or another tool that could be used to extract sequences and to later feed back into downstream circRNAprofiler functions which require the 'targetsFTS_gr' object?

Any help on this general issue I have will be of a great help.

Thanks,
Ketan

**Comment by Aufiero** [comment]
Which genome are you working with? Did you check if a BSgenome package for your organism already exists?

**Comment by Aufiero** [comment]
circRNAprofiler can only be used for now with BSgenome.

**Comment by 1kabutar** [comment]
Salamander genome which is 20gb. Yes, I checked they don't have it there.
I will try to contact the BSgenome people too for the 64-bit support so can accomodate large genomes.
